scheduler/src/cook/mesos/rebalancer.clj

```diff
@@ -382,12 +382,22 @@
             :to-make-room-for pending-job)]
     [state nil]))
 
+(defn reserve-hosts! [state preemption-decisions]
+  (let [multiple-tasks (filter #(< 1 (count (:task %))) preemption-decisions)
+        reservations (into {} (map (fn [d] [(:job/uuid (:to-make-room-for d))
+                                            (:hostname d)])
+                                   multiple-tasks))]
+    (swap! state (fn [{:keys [jobs-launched]}]
+                   {:reservations (apply dissoc reservations jobs-launched)
```

Comment:
Nitpick, should we rename `reservations` to `job-uuid->reserved-host`

Reply:
If only we had a type system to tell us these kind of things...

Done